frontend/src/app/user-profile/user-profile.component.ts

```diff
@@ -27,6 +27,15 @@ export class UserProfileComponent implements OnInit {
     private fb: FormBuilder,
   ) { }
 
+  validatePhone() {
+    return [
+      Validators.required,
+      Validators.minLength(9),
+      Validators.maxLength(11),
+      Validators.pattern(/^[0-9]{9}$|^[0-9]{3}[-\s][0-9]{3}[-\s][0-9]{3}$/)
+    ]
+  }
+
```

Comment:
This function is the same in user-profile.component and contac.component. I think it would be nice to make it reusable in utils or smth like this.

Reply:
IMHO it is reasonable to write a test for a new method you've added in auth.service and maybe test to check whether phone_no field in the contact form is valid/invalid depends on the value provided in a form.